Infrastructure/Repositories/ProjectRepository.cs

```diff
@@ -2,83 +2,59 @@
-        },
-        new ProjectModel
-        {
-            Id = Guid.NewGuid(),
-            Name = "Project 2",
-            Description = "Description 2",
-            StartDate = DateOnly.MinValue,
-            Tasks = []
-        }
-    ];
-
     public async Task<IEnumerable<ProjectModel>> GetAllAsync(ProjectSearchParams projectSearchParams)
     {
-        return await Task.Run(() =>
+        var allProjects = dbContext.Projects.AsQueryable();
+        if (!string.IsNullOrEmpty(projectSearchParams.Name))
         {
-            var allProjects = _projects.AsQueryable();
-            if (!string.IsNullOrEmpty(projectSearchParams.Name))
-            {
-                allProjects = allProjects
-                    .Where(p => p.Name.Contains(projectSearchParams.Name, StringComparison.OrdinalIgnoreCase));
-            }
+            allProjects = allProjects
+                .Where(p => p.Name.Contains(projectSearchParams.Name, StringComparison.OrdinalIgnoreCase));
+        }
 
-            if (!string.IsNullOrEmpty(projectSearchParams.Description))
-            {
-                allProjects = allProjects
-                    .Where(p => p.Description.Contains(projectSearchParams.Description,
-                        StringComparison.OrdinalIgnoreCase));
-            }
+        if (!string.IsNullOrEmpty(projectSearchParams.Description))
+        {
+            allProjects = allProjects
+                .Where(p => p.Description.Contains(projectSearchParams.Description,
+                    StringComparison.OrdinalIgnoreCase));
+        }
 
-            if (projectSearchParams.StartDate.HasValue)
-            {
-                allProjects = allProjects
-                    .Where(p => p.StartDate == projectSearchParams.StartDate.Value);
-            }
+        if (projectSearchParams.StartDate.HasValue)
+        {
+            allProjects = allProjects
+                .Where(p => p.StartDate == projectSearchParams.StartDate.Value);
+        }
 
-            if (projectSearchParams.StartDateFrom.HasValue)
-            {
-                allProjects = allProjects
-                    .Where(p => p.StartDate >= projectSearchParams.StartDateFrom.Value);
-            }
+        if (projectSearchParams.StartDateFrom.HasValue)
+        {
+            allProjects = allProjects
+                .Where(p => p.StartDate >= projectSearchParams.StartDateFrom.Value);
+        }
 
-            if (projectSearchParams.StartDateTo.HasValue)
-            {
-                allProjects = allProjects
-                    .Where(p => p.StartDate <= projectSearchParams.StartDateTo.Value);
-            }
+        if (projectSearchParams.StartDateTo.HasValue)
+        {
+            allProjects = allProjects
+                .Where(p => p.StartDate <= projectSearchParams.StartDateTo.Value);
+        }
 
-            return allProjects;
-        });
+        allProjects = allProjects.Include(p => p.Tasks);
+        return await allProjects.ToListAsync();
```

Comment:
Why do we need to convert it to the list?

Reply:
because "Type 'System.Linq.IQueryable<Domain.Models.ProjectModel>' is not awaitable";
Thats an error i got, if i delete ToListAsync();

Reply:
Why do you need to await it if it's not awaitable?

Reply:
ahh okay